### PR TITLE
releng - fix poetrypkg gen-setup command

### PR DESCRIPTION
**Pull request by thisisshi**
\+ pkg rebase, tried to do a pkg freeze but tencentcloud had some conflicts and mailer had a importlib-metadata conflict

### CODE REVIEW

File: tools/dev/poetrypkg.py [hunk 1]
```diff
@@ -198,15 +198,19 @@ def resolve_source_deps(poetry, package, reqs, frozen=False):
 
     from poetry.core.packages.dependency import Dependency
 
-    dep_map = {d['name']: d for d in poetry.locker.lock_data['package']}
+    # normalize deps by lowercasing all the keys
+    dep_map = {d['name'].lower(): d for d in poetry.locker.lock_data['package']}
```
Comment (thisisshi) [inline tools/dev/poetrypkg.py:202]: e.g. PyYAML vs pyyaml

File: tools/c7n_org/setup.py [hunk 2]
```diff
@@ -48,9 +48,9 @@
...
     'long_description_content_type': 'text/markdown',
     'author': 'Cloud Custodian Project',
-    'author_email': None,
-    'maintainer': None,
-    'maintainer_email': None,
+    'author_email': 'None',
```
Comment (ajkerrigan) [inline tools/c7n_org/setup.py:51]: Seems a bit weird to stringify the `None` here especially for the optional maintainer parameters. That's an upstream concern though, and not a blocker. Just looked odd at a glance.

**Review by ajkerrigan — APPROVED** [review]
Looks good, thanks :+1:

File: tools/dev/poetrypkg.py (hunk 1 above)
Comment (ajkerrigan) [reply tools/dev/poetrypkg.py:202]: Makes sense, mirroring upstream changes :+1:

File: tools/c7n_awscc/setup.py [hunk 3]
```diff
@@ -46,9 +46,9 @@
     'long_description': '\n# Custodian AWS Cloud Control Provider\n\n\n',
     'long_description_content_type': 'text/markdown',
     'author': 'Cloud Custodian Project',
-    'author_email': None,
-    'maintainer': None,
-    'maintainer_email': None,
+    'author_email': 'None',
```
Comment (kapilt) [inline tools/c7n_awscc/setup.py:49]: this seems suspect

File: tools/dev/poetrypkg.py (hunk 1 above)
Comment (kapilt) [reply tools/dev/poetrypkg.py:202]: which upstream changes? this feels very odd, mutating locked package names.

Comment (ajkerrigan) [reply tools/dev/poetrypkg.py:202]: There have been a few changes related to package name normalization/canonicalization over time. Specifically, this made me think of the recent change from `pyyaml` --> `PyYAML` in `poetry.lock` you can see [here](https://github.com/cloud-custodian/cloud-custodian/pull/7844/files/89619ee3d9d8797d9ea37288d72b6116d4f74c16#diff-f53a023eedfa3fbf2925ec7dc76eecdc954ea94b7e47065393dbad519613dc89L670-R670):

```  
 233   │  [[package]]
 234   │ -name = "pyyaml"
 235   │ +name = "PyYAML"
 236   │  version = "6.0"
 237   │  description = "YAML parser and emitter for Python"
 238   │  category = "main"
 239   │ @@ -676,7 +676,7 @@ python-versions = ">=3.6"
```

**Pull request merged**